src/s3/commands/presignedURL.ts

```diff
@@ -38,36 +45,124 @@ export async function presignedURLCommand(
         })
 
         await copyToClipboard(url, 'URL', window, env)
+        vscode.window.showInformationMessage(`A presigned URL has been copied to your clipboard`)
     })
 }
 
-export async function promptTime(fileName: string, window = Window.vscode()): Promise<number> {
-    const timeStr = await window.showInputBox({
-        value: '15',
-        prompt: localize(
-            'AWS.s3.promptTime.prompt',
-            'Specify the time (minutes) until URL will expire for path: {0}',
-            fileName
-        ),
-        placeHolder: localize('AWS.s3.promptTime.placeHolder', 'Defaults to 15 minutes'),
-        validateInput: validateTime,
-    })
+function validateTime(time: string): string | undefined {
+    const number = Number(time)
+
+    if (isNaN(Number(time)) || !Number.isSafeInteger(number) || Number(time) <= 0) {
+        return invalidNumberWarning
+    }
+
+    return undefined
+}
+
+export interface PresignedUrlWizardState {
+    region: string
+    signedUrlParams: SignedUrlRequest
+}
+
+export class PresignedUrlWizard extends Wizard<PresignedUrlWizardState> {
+    constructor(node?: S3FileNode) {
+        super({
+            initState: {
+                region: node?.bucket.region,
+                signedUrlParams: node ? ({ bucketName: node.bucket.name } as SignedUrlRequest) : undefined,
```

Comment:
somewhere, perhaps in a `/** ... */` comment on this param, we need to explain what's going on. It's not obvious from any of the function names or anything AFAICT, that this wizard has two different behaviors that are decided by  (what?)

In the future, try to anticipate the perspective of other developers, this kind of question they will ask as they read the code.

Reply:
added a comment